client/src/app/api/api/api.service.ts

```diff
@@ -115,4 +153,64 @@ export class ApiService {
       }
     );
   }
+  // This is to send the POST request to the server
+  private post(endpoint: string, componentType: string, name: string, observe: any = 'body', reportProgress: boolean = false): any {
```

Comment:
why do both post methods contain virtually the same code except for the parameters which are send to the server?   please design a data model to send to the server to enable reuse of the post method.